source/core/slang-writer.h

```diff
@@ -45,7 +45,7 @@ class BaseWriter : public ISlangWriter, public RefObject
     // ISlangUnknown
     SLANG_REF_OBJECT_IUNKNOWN_QUERY_INTERFACE
     SLANG_REF_OBJECT_IUNKNOWN_ADD_REF
-    SLANG_NO_THROW uint32_t SLANG_MCALL release() { return (m_flags & WriterFlag::IsStatic) ? 1 : (uint32_t)releaseReference(); }
+    SLANG_NO_THROW uint32_t SLANG_MCALL release() { return (m_flags & WriterFlag::IsStatic) ? (uint32_t)decreaseReference() : (uint32_t)releaseReference(); }
```

Comment:
Can you explain this change? I don't know what `decreaseReference()` is (other than the obvious answer that it probably decrements the reference count), and I'm also not clear on why a writer with static ownership would need/want to mess with the reference count at all.

Reply:
It was not super important, but when I thought a little more about my 'make static' work around, it breaks some of the semantics of what release means. That what I really care about is that the object isn't deleted if the refcount hits zero. That if I leave the ref counting running (without the delete), it makes the values returned make more sense, and for example I can verify if the refcount is 0 - to check for lingering refs and the like when debugging. This may or may not be right depending on usage (say I had made static, and multiple threads using and refing, which would be ok - well then the ref would still be wrong). I guess the short answer is that this is 'more right'.